service/src/lib.rs

```diff
@@ -44,8 +45,21 @@ pub use primitives::Blake2Hasher;
 pub use sp_runtime::traits::ProvideRuntimeApi;
 pub use sc_network::specialization::NetworkSpecialization;
 pub use chain_spec::ChainSpec;
+#[cfg(not(target_os = "unknown"))]
 pub use consensus::run_validation_worker;
 
+/// Wrap a futures01 executor as a futures03 spawn.
+#[derive(Clone)]
+pub struct WrappedExecutor<T>(pub T);
```

Comment:
There's https://docs.rs/futures/0.3.1/futures/compat/struct.Executor01As03.html

Reply:
I don't think we meet the trait bounds:

<img width="696" alt="Screen Shot 2019-12-09 at 5 43 38 PM" src="https://user-images.githubusercontent.com/13566135/70454863-83221480-1aab-11ea-9768-91d3712ddbce.png">